pushnotifications-integration-tests/src/androidTest/java/com/example/pushnotificationsintegrationtests/MultiInstanceSupportTest.kt

```diff
@@ -0,0 +1,338 @@
+    pni2.setUserId("bob", tokenProvider2, object: BeamsCallback<Void, PusherCallbackError> {
+      override fun onSuccess(vararg values: Void) {
+        successWasCalled2 = true
+      }
+
+      override fun onFailure(error: PusherCallbackError) {
+        failureWasCalled2 = true
+      }
+    })
+
+    // A device ID should have been stored
+    assertStoredDeviceIdIsNotNull(instanceId1)
+    assertStoredDeviceIdIsNotNull(instanceId2)
+
+    val storedDeviceId1 = getStoredDeviceId(instanceId1)
+    val storedDeviceId2 = getStoredDeviceId(instanceId2)
+
+    // and the stored ids should match the server one
+    assertNotNull(errolClient1.getDevice(storedDeviceId1!!))
+    assertNotNull(errolClient2.getDevice(storedDeviceId2!!))
+
+    // check the server has the same interests
+    await.atMost(3, TimeUnit.SECONDS) until {
+      errolClient1.getDevice(storedDeviceId1)?.userId == "alice"
+    }
+    await.atMost(3, TimeUnit.SECONDS) until {
+      errolClient2.getDevice(storedDeviceId2)?.userId == "bob"
+    }
+
+    // make sure the callbacks were correctly triggered
+    assertTrue(successWasCalled1)
+    assertFalse(failureWasCalled1)
+    assertTrue(successWasCalled2)
+    assertFalse(failureWasCalled2)
+  }
+
+  @Test
+  fun setOnDeviceInterestsChangedListenerNotAffectTheOther() {
+    // Start both instances
+    val pni1 = PushNotificationsInstance(context, instanceId1)
+    pni1.addDeviceInterest("hello")
+    pni1.start()
+
+    val pni2 = PushNotificationsInstance(context, instanceId2)
+    pni2.start()
+
+    // A device ID should have been stored
+    assertStoredDeviceIdIsNotNull(instanceId1)
+    assertStoredDeviceIdIsNotNull(instanceId2)
+
+    val storedDeviceId1 = getStoredDeviceId(instanceId1)
+    val storedDeviceId2 = getStoredDeviceId(instanceId2)
+
+    // and the stored ids should match the server one
+    assertNotNull(errolClient1.getDevice(storedDeviceId1!!))
+    assertNotNull(errolClient2.getDevice(storedDeviceId2!!))
+
+    var setOnSubscriptionsChangedListenerCalledCount1 = 0
+    var lastSetOnSubscriptionsChangedListenerCalledWithInterests1: Set<String>? = null
+    pni1.setOnDeviceInterestsChangedListener(object : SubscriptionsChangedListener {
+      override fun onSubscriptionsChanged(interests: Set<String>) {
+        setOnSubscriptionsChangedListenerCalledCount1++
+        lastSetOnSubscriptionsChangedListenerCalledWithInterests1 = interests
+      }
+    })
+    var setOnSubscriptionsChangedListenerCalledCount2 = 0
+    var lastSetOnSubscriptionsChangedListenerCalledWithInterests2: Set<String>? = null
+    pni2.setOnDeviceInterestsChangedListener(object : SubscriptionsChangedListener {
+      override fun onSubscriptionsChanged(interests: Set<String>) {
+        setOnSubscriptionsChangedListenerCalledCount2++
+        lastSetOnSubscriptionsChangedListenerCalledWithInterests2 = interests
+      }
+    })
+
+    // this will cause it to receive the initial interest set of {"hello"}
```

Comment:
this comment isn't true? it's an initial interest set of {"potatoes"} - and we were using single versions of things so it should just be one potato 🥔

Reply:
I think this was just a case of blind copy-pasting